convert2rhel/unit_tests/applock_test.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+
+import pytest
+
+from convert2rhel import applock
+
+
+@pytest.fixture
+def tmp_lock(monkeypatch, tmp_path):
+    monkeypatch.setattr(applock, "_DEFAULT_LOCK_DIR", str(tmp_path))
+    alock = applock.ApplicationLock("convert2rhelTEST")
+    return alock
+
+
+def test_applock_context_manager(monkeypatch, tmp_path):
+    monkeypatch.setattr(applock, "_DEFAULT_LOCK_DIR", str(tmp_path))
+    pidfile = "/non/existent/file"
```

Comment:
> Incorrect: if the context manager **enter** function fails, pidfile may not be set before it is tested.

Actually, I think codeQL is correct here? If the `__enter__()` function fails, it will raise an exception.  Since we aren't catching the exception, the lines where we use pidfile will never be executed.